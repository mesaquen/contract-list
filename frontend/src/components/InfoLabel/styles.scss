.infoLabel--container {
  display: inline-flex;
  align-items: center;
}

.infoLabel--icon {
  margin-left: 0.5rem;
  display: flex;
  flex-grow: 0;
}
