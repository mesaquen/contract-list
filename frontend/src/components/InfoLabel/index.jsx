import React from 'react'
import PropTypes from 'prop-types'
import InfoIcon from '../InfoIcon'
import Typography from '../Typography'
import './styles.scss'

const InfoLabel = ({ label, info, ...props }) => {
  return (
    <div className="infoLabel--container" {...props}>
      <Typography size={Typography.S} variant="secondary" bold>
        {label}
      </Typography>
      <InfoIcon size="1.5rem" info={info} className="infoLabel--icon" />
    </div>
  )
}

InfoLabel.propTypes = {
  label: PropTypes.string,
  info: PropTypes.string,
}

export default InfoLabel
