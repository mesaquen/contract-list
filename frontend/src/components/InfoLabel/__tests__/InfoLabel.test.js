import { cleanup, render } from '@testing-library/react'
import InfoLabel from '../index'

beforeEach(cleanup)

describe('InfoLabel', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<InfoLabel label="sample" info="info" />)
    expect(asFragment()).toMatchSnapshot()
  })
})
