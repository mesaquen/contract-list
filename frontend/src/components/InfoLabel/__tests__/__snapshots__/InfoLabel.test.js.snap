// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoLabel should render correctly 1`] = `
<DocumentFragment>
  <div
    class="infoLabel--container"
  >
    <span
      class="typo--s typo--bold typo--text-secondary"
    >
      sample
    </span>
    <span
      class="infoLabel--icon"
      title="info"
    >
      <svg
        class="infoIcon--icon"
        fill="currentColor"
        height="1.5rem"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1.5rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
        />
      </svg>
    </span>
  </div>
</DocumentFragment>
`;
